### PR TITLE
Jerk - per axis maximum limit

### PR DESCRIPTION
**Pull request by vovodroid**
Currently both X/Y jerks are limited by lower jerk value from **Jerk limitation** settings. But for large bedslingers Y axis could have much less jerk then X due to weight. This PR set each axis jerk according to corresponding limit.

image

image

image

### CODE REVIEW

File: src/libslic3r/GCodeWriter.cpp [hunk 1]
```diff
@@ -222,9 +224,15 @@ std::string GCodeWriter::set_acceleration_internal(Acceleration type, unsigned i
 
 std::string GCodeWriter::set_jerk_xy(double jerk)
 {
+    double jerk_x = jerk;
+    double jerk_y = jerk;
+    
     // Clamp the jerk to the allowed maximum.
-    if (m_max_jerk > 0 && jerk > m_max_jerk)
-        jerk = m_max_jerk;
+    if (m_max_jerk_x > 0 && jerk > m_max_jerk_x)
```
Comment (SoftFever) [inline src/libslic3r/GCodeWriter.cpp:231]: We still need to cap jerk, it's used by Klipper in below.

File: src/libslic3r/GCodeWriter.cpp [hunk 2]
```diff
@@ -249,6 +257,9 @@ std::string GCodeWriter::set_jerk_xy(double jerk)
 
 std::string GCodeWriter::set_accel_and_jerk(unsigned int acceleration, double jerk)
 {
+    double jerk_x = jerk;
```
Comment (SoftFever) [inline src/libslic3r/GCodeWriter.cpp:260]: jerk_x/y is not supported by klipper, and these two valriables are not used here.
We should revert changes in this function

**Review by SoftFever — CHANGES_REQUESTED** [review]

File: src/libslic3r/GCodeWriter.cpp (hunk 2 above)
Comment (vovodroid) [reply src/libslic3r/GCodeWriter.cpp:260]: Looks reasonable?
```
    // Clamp the jerk to the allowed maximum.
    if (m_max_jerk_x > 0 && jerk > m_max_jerk_x)
        jerk = m_max_jerk_x;
    if (m_max_jerk_y > 0 && jerk > m_max_jerk_y)
        jerk = m_max_jerk_y;
```

File: src/libslic3r/GCodeWriter.cpp (hunk 1 above)
Comment (vovodroid) [reply src/libslic3r/GCodeWriter.cpp:231]: Like this?
```
    m_last_jerk = jerk;
    jerk = std::min(jerk, std::min(jerk_x, jerk_y));
```

Comment (SoftFever) [reply src/libslic3r/GCodeWriter.cpp:231]: It's better to move the logic inside following block respectively to make it cleaner as Klipper don't use jerk_x/y.
```
if(FLAVOR_IS(gcfKlipper))
        gcode << "SET_VELOCITY_LIMIT SQUARE_CORNER_VELOCITY=" << jerk;
    else
        gcode << "M205 X" << jerk_x << " Y" << jerk_y;
```

File: src/libslic3r/GCodeWriter.cpp (hunk 2 above)
Comment (SoftFever) [reply src/libslic3r/GCodeWriter.cpp:260]: yes

Comment (SoftFever) [reply src/libslic3r/GCodeWriter.cpp:265]: Can we remove this two unused lines?

Comment (vovodroid) [reply src/libslic3r/GCodeWriter.cpp:265]: They are used, we still need to cap single jerk to slowest axis.

Comment (vovodroid) [reply src/libslic3r/GCodeWriter.cpp:265]: I updated PR branch.

Comment (SoftFever) [reply src/libslic3r/GCodeWriter.cpp:265]: I see it's still here.
These two variables are not used in this function, so they can be deleted.

Comment (vovodroid) [reply src/libslic3r/GCodeWriter.cpp]: Sorry, I didn't get you are talking about variable declarations in `set_accel_and_jerk`. Removed.

**Pull request merged**